upgrade.md

```diff
@@ -29,65 +25,56 @@ Provided you have suitable udev rules in play, this will already do everything y
+
 Plug in the Black Magic Probe into your computer and run the following command:
 
 ```bash
 sudo dfu-util -d 1d50:6018,:6017 -s 0x08002000:leave -D blackmagic-native.bin
 ```
 
-You will need to install [`dfu-util`](http://dfu-util.sourceforge.net/) package. On MacOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
+To upgrade non-native hardware see the readme of the different [platforms on GitHub](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms).
 
 ```{note}
 If `dfu-util` fails to switch your BMP into bootloader mode, or you feel like you might have **bricked** your BMP, you can also plug in your BMP while holding down the button. This will force the BMP to stay in the bootloader on power up.
 ```
 
-## Windows
+### Using stm32-mem.py
 
-### bmputil
-
-Download or compile a Black Magic Probe (BMP) firmware.
-
-Clone and build, or directly install the bmputil tool:
+An alternative solution is using `stm32_mem.py` script included in the Black Magic Debug source repository. It is not the recommended way of updating the Black Magic Probe as `dfu-util` is easier to install without the need to clone the Black Magic Debug repository and all the associated source code.
 
 ```bash
-cargo install --git https://github.com/blackmagic-debug/bmputil
+cd blackmagic && scripts/stm32_mem.py blackmagic.bin
 ```
 
-Plug in your Black Magic Probe and run the following command:
+### stlink-tool
+
+An ST-Link can be used to upate the firmware using [stlink-tool](<https://github.com/jeanthom/stlink-tool>).
 
 ```bash
-bmputil flash blackmagic-native.elf
+stlink-tools blackmagic.bin
 ```
 
-bmputil will automatically take care of your driver needs as appropriate.
+## Windows (specific)
+
+### bmputil
+For Windows, the bmputil instructions don't change.
+
+bmputil will automatically take care of the needed drivers.
 Do **not** run Zadig on the DFU endpoint if using bmputil.
 
 ### dfu-util
 
-Download or compile a Black Magic Probe (BMP) firmware.
+You will need to download the Windows release of [`dfu-util`](http://dfu-util.sourceforge.net/) package.
 
 Plug in the Black Magic Probe into your computer and run the following command:
 
 ```bash
 dfu-util.exe -d 1d50:6018,:6017 -s 0x08002000:leave -D blackmagic.bin
 ```
 
-You will need to download the Windows release of [`dfu-util`](http://dfu-util.sourceforge.net/) package.
-
 You may need use [Zadig](https://tracker.iplocation.net/icsj/) to install the WinUSB driver. Both PID 0x6017 and 0x6018 need to be known to  [Zadig](https://tracker.iplocation.net/icsj/).
 
 ```{note}
 On first run Zadig may not see the 6017 PID. Once the dfu exe runs and detaches the device, it should appear in Zadig. Install the WinUSB driver on it and re-run the dfu exe.
 ```
-
-```{note}
```

Comment:
This note is needed somewhere around the firmware update section to document how the user button works when held and the probe plugged in

Reply:
This comment is addressed in the latest commit to `upgrade.md`.